app/Actions/Import/Exec.php

```diff
@@ -232,7 +239,28 @@ public function do(
 			foreach ($files as $file) {
 				$this->assertImportNotCancelled();
 				// Reset the execution timeout for every iteration.
-				set_time_limit(ini_get('max_execution_time'));
+				try {
+					\Safe\set_time_limit((int) \Safe\ini_get('max_execution_time'));
+				} catch (\Throwable) {
+					// ! If we do not catch this one this throws this error during tests:
+					/**
+					 * 1) Tests\Feature\PhotosAddTest::testImport
+					 * Expected response status code [204] but received 422.
+					 *
+					 * The following errors occurred during the request:
+					 *
+					 * {
+					 *     "message": "The given data was invalid.",
+					 *     "errors": {
+					 *         "photoIDs": [
+					 *             "The photo i ds field is required."
+					 *         ]
+					 *     }
+					 * }
+					 *
+					 * Failed asserting that 204 is identical to 422.
+					 */
+				}
```

Comment:
It is somehow comprehensible that `set_time_limit` throws an exception, if extended one's own time limit is forbidden and of course, we want to proceed in this case nonetheless.

However, I do not understand why this eventually leads to a 422 error, in particular as `photo_id` is given and is not related to this exception. (This sound like another bug.)

Anyway, I would remove that long comment and replace it by something which really explains why we catch the exception. A failing test is a bad explanation and sounds like a cheap excuse.

```suggestion
				try {
					\Safe\set_time_limit((int) \Safe\ini_get('max_execution_time'));
				} catch (\Throwable) {
					// if resetting the time limit is forbidden, we proceed nonetheless as long as we can
				}
```

However, I wonder if there is any installation out there which actually allows to reset the time limit by a script. This sounds like something which should be always forbidden, because it constitutes a security problem. This way a script could easily mount DoD attacks.

Maybe we should removed these lines of code completely?

Reply:
Fixed in https://github.com/LycheeOrg/Lychee/pull/1336/commits/67962c91318de2146bf22694873143df13b68baf.